scripts/preprocess-docs.js

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env node
+
+/**
+ * Pre-processes Mocha's documentation
+ * 1. Update TOC
+ * 2. Dump `--help` output into "Command-Line Usage" section
+ */
+
+/**
+ * The CLI included with markdown-toc doesn't support `bullets`
+ * and `maxdepth` options, so that's why this exists.
+ */
+
+'use strict';
+
+const toc = require('markdown-toc');
+const miss = require('mississippi');
+const {exec} = require('child_process');
+const fs = require('fs');
+const {stripIndents} = require('common-tags');
+const path = require('path');
+
+const docsFilepath = path.join(__dirname, '..', 'docs', 'index.md');
+
+/**
+ * Inserts/updates a table of contents into the documentation
+ * @returns Promise<void>
+ */
+function updateTOC() {
+  return new Promise((resolve, reject) => {
+    console.error(`Updating TOC in ${docsFilepath}...`);
+
+    miss.pipe(
+      fs.createReadStream(docsFilepath, 'utf8'),
+      miss.concat(input =>
+        fs.writeFileSync(
+          docsFilepath,
+          toc
+            .insert(String(input), {
+              bullets: '-',
+              maxdepth: 2
+            })
+            .replace(/\n\n$/, '\n')
+        )
+      ),
+      err => {
+        if (err) {
+          return reject(err);
+        }
+        console.error(`Updated TOC in ${docsFilepath}.`);
```

Comment:
If we're in error section, did we update the TOC or fail to do so?

Reply:
we're not in an error-handler at this point?

Reply:
Guess I misunderstood what (L46) `err => {` meant. We appear to be in an error handler.